File: bin/Debug/net5.0/v4.4.2/lgcy-on-lgcy/examples/t_api-captcha-webforms-web.security.membership/vbnet/App_Code/MockMembershipProvider.vb
Imports Microsoft.VisualBasic

Namespace WebFormsMembershipCaptchaExampleVBNet

    Public Class MockMembershipProvider
        Inherits System.Web.Security.MembershipProvider
        Public Sub New()
        End Sub

        Public Overrides Property ApplicationName() As String
            Get
                Return "/"
            End Get
            Set(ByVal value As String)
            End Set
        End Property

        Public Overrides Function ChangePassword(ByVal username As String, ByVal oldPassword As String, ByVal newPassword As String) As Boolean
            Return True
        End Function

        Public Overrides Function ChangePasswordQuestionAndAnswer(ByVal username As String, ByVal password As String, ByVal newPasswordQuestion As String, ByVal newPasswordAnswer As String) As Boolean
            Return True
        End Function

        Public Overrides Function CreateUser(ByVal username As String, ByVal password As String, ByVal email As String, ByVal passwordQuestion As String, ByVal passwordAnswer As String, ByVal isApproved As Boolean,
         ByVal providerUserKey As Object, ByRef status As MembershipCreateStatus) As MembershipUser
            Dim user As New MembershipUser("MockMembershipProvider", username, Nothing, email, passwordQuestion, Nothing,
             isApproved, False, DateTime.Now, DateTime.Now, DateTime.Now, DateTime.Now,
             DateTime.Now)
            status = MembershipCreateStatus.Success
            Return user
        End Function

        Public Overrides Function DeleteUser(ByVal username As String, ByVal deleteAllRelatedData As Boolean) As Boolean
            Return True
        End Function

        Public Overrides ReadOnly Property EnablePasswordReset() As Boolean
            Get
                Return True
            End Get
        End Property

        Public Overrides ReadOnly Property EnablePasswordRetrieval() As Boolean
            Get
                Return True
            End Get
        End Property

        Public Overrides Function FindUsersByEmail(ByVal emailToMatch As String, ByVal pageIndex As Integer, ByVal pageSize As Integer, ByRef totalRecords As Integer) As MembershipUserCollection
            totalRecords = 0
            Return New MembershipUserCollection()
        End Function

        Public Overrides Function FindUsersByName(ByVal usernameToMatch As String, ByVal pageIndex As Integer, ByVal pageSize As Integer, ByRef totalRecords As Integer) As MembershipUserCollection
            totalRecords = 0
            Return New MembershipUserCollection()
        End Function

        Public Overrides Function GetAllUsers(ByVal pageIndex As Integer, ByVal pageSize As Integer, ByRef totalRecords As Integer) As MembershipUserCollection
            totalRecords = 0
            Return New MembershipUserCollection()
        End Function

        Public Overrides Function GetNumberOfUsersOnline() As Integer
            Return 0
        End Function

        Public Overrides Function GetPassword(ByVal username As String, ByVal answer As String) As String
            Return "password"
        End Function

        Public Overrides Function GetUser(ByVal username As String, ByVal userIsOnline As Boolean) As MembershipUser
            Dim user As New MembershipUser("MockMembershipProvider", username, Nothing, Nothing, Nothing, Nothing,
             True, False, DateTime.Now, DateTime.Now, DateTime.Now, DateTime.Now,
             DateTime.Now)
            Return user
        End Function

        Public Overrides Function GetUser(ByVal providerUserKey As Object, ByVal userIsOnline As Boolean) As MembershipUser
            Dim user As New MembershipUser("MockMembershipProvider", Nothing, Nothing, Nothing, Nothing, Nothing,
             True, False, DateTime.Now, DateTime.Now, DateTime.Now, DateTime.Now,
             DateTime.Now)
            Return user
        End Function

        Public Overrides Function GetUserNameByEmail(ByVal email As String) As String
            Return "username"
        End Function

        Public Overrides ReadOnly Property MaxInvalidPasswordAttempts() As Integer
            Get
                Return 0
            End Get
        End Property

        Public Overrides ReadOnly Property MinRequiredNonAlphanumericCharacters() As Integer
            Get
                Return 0
            End Get
        End Property

        Public Overrides ReadOnly Property MinRequiredPasswordLength() As Integer
            Get
                Return 0
            End Get
        End Property

        Public Overrides ReadOnly Property PasswordAttemptWindow() As Integer
            Get
                Return 0
            End Get
        End Property

        Public Overrides ReadOnly Property PasswordFormat() As MembershipPasswordFormat
            Get
                Return MembershipPasswordFormat.Clear
            End Get
        End Property

        Public Overrides ReadOnly Property PasswordStrengthRegularExpression() As String
            Get
                Return ""
            End Get
        End Property

        Public Overrides ReadOnly Property RequiresQuestionAndAnswer() As Boolean
            Get
                Return False
            End Get
        End Property

        Public Overrides ReadOnly Property RequiresUniqueEmail() As Boolean
            Get
                Return True
            End Get
        End Property

        Public Overrides Function ResetPassword(ByVal username As String, ByVal answer As String) As String
            Return "password"
        End Function

        Public Overrides Function UnlockUser(ByVal userName As String) As Boolean
            Return True
        End Function

        Public Overrides Sub UpdateUser(ByVal user As MembershipUser)
        End Sub

        Public Overrides Function ValidateUser(ByVal username As String, ByVal password As String) As Boolean
            Dim result As Boolean
            result = False

            If username.Trim().Length > 4 And password.Trim().Length > 4 Then
                result = True
            End If

            Return result
        End Function
    End Class

End Namespace

File: bin/Debug/net5.0/v4.4.2/lgcy-on-lgcy/examples/t_api-captcha-webforms-web.security.membership/csharp/App_Code/MockMembershipProvider.cs
using System;
using System.Data;
using System.Configuration;
using System.Web;
using System.Web.Security;
using System.Web.UI;
using System.Web.UI.HtmlControls;
using System.Web.UI.WebControls;
using System.Web.UI.WebControls.WebParts;

namespace WebFormsMembershipCaptchaExampleCSharp
{
    /// <summary>
    /// A mock MembershipProvider, doesn't actually work but makes the example simpler, 
    /// since it's functionality is irrelevant to Captcha integration
    /// </summary>
    public class MockMembershipProvider : System.Web.Security.MembershipProvider
    {
        public MockMembershipProvider()
        {
        }

        public override string ApplicationName
        {
            get
            {
                return "/";
            }
            set
            {
            }
        }

        public override bool ChangePassword(string username, string oldPassword, string newPassword)
        {
            return true;
        }

        public override bool ChangePasswordQuestionAndAnswer(string username, string password, string newPasswordQuestion, string newPasswordAnswer)
        {
            return true;
        }

        public override MembershipUser CreateUser(string username, string password, string email, string passwordQuestion, string passwordAnswer, bool isApproved, object providerUserKey, out MembershipCreateStatus status)
        {
            MembershipUser user = new MembershipUser("MockMembershipProvider", username, null, email, passwordQuestion, null, isApproved, false, DateTime.Now, DateTime.Now, DateTime.Now, DateTime.Now, DateTime.Now);
            status = MembershipCreateStatus.Success;
            return user;
        }

        public override bool DeleteUser(string username, bool deleteAllRelatedData)
        {
            return true;
        }

        public override bool EnablePasswordReset
        {
            get { return true; }
        }

        public override bool EnablePasswordRetrieval
        {
            get { return true; }
        }

        public override MembershipUserCollection FindUsersByEmail(string emailToMatch, int pageIndex, int pageSize, out int totalRecords)
        {
            totalRecords = 0;
            return new MembershipUserCollection();
        }

        public override MembershipUserCollection FindUsersByName(string usernameToMatch, int pageIndex, int pageSize, out int totalRecords)
        {
            totalRecords = 0;
            return new MembershipUserCollection();
        }

        public override MembershipUserCollection GetAllUsers(int pageIndex, int pageSize, out int totalRecords)
        {
            totalRecords = 0;
            return new MembershipUserCollection();
        }

        public override int GetNumberOfUsersOnline()
        {
            return 0;
        }

        public override string GetPassword(string username, string answer)
        {
            return "password";
        }

        public override MembershipUser GetUser(string username, bool userIsOnline)
        {
            MembershipUser user = new MembershipUser("MockMembershipProvider", username, null, null, null, null, true, false, DateTime.Now, DateTime.Now, DateTime.Now, DateTime.Now, DateTime.Now);
            return user;
        }

        public override MembershipUser GetUser(object providerUserKey, bool userIsOnline)
        {
            MembershipUser user = new MembershipUser("MockMembershipProvider", null, null, null, null, null, true, false, DateTime.Now, DateTime.Now, DateTime.Now, DateTime.Now, DateTime.Now);
            return user;
        }

        public override string GetUserNameByEmail(string email)
        {
            return "username";
        }

        public override int MaxInvalidPasswordAttempts
        {
            get { return 0; }
        }

        public override int MinRequiredNonAlphanumericCharacters
        {
            get { return 0; }
        }

        public override int MinRequiredPasswordLength
        {
            get { return 0; }
        }

        public override int PasswordAttemptWindow
        {
            get { return 0; }
        }

        public override MembershipPasswordFormat PasswordFormat
        {
            get { return MembershipPasswordFormat.Clear; }
        }

        public override string PasswordStrengthRegularExpression
        {
            get { return ""; }
        }

        public override bool RequiresQuestionAndAnswer
        {
            get { return false; }
        }

        public override bool RequiresUniqueEmail
        {
            get { return true; }
        }

        public override string ResetPassword(string username, string answer)
        {
            return "password";
        }

        public override bool UnlockUser(string userName)
        {
            return true;
        }

        public override void UpdateUser(MembershipUser user)
        {
        }

        public override bool ValidateUser(string username, string password)
        {
            bool result = false;
            if (username.Trim().Length > 4 && password.Trim().Length > 4)
            {
                result = true;
            }
            return result;
        }
    }
}
